### PR TITLE
Example refactoring

### PR DESCRIPTION
**Pull request by inetic**
To the new Routing API

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/571)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: examples/simple_key_value_store.rs [hunk 1]
```diff
@@ -43,11 +43,12 @@ extern crate maidsafe_sodiumoxide as sodiumoxide;
 extern crate crust;
 extern crate routing;
 
-use core::iter::FromIterator;
+//use core::iter::FromIterator;
 use std::io;
 use std::net::SocketAddr;
 use std::str::FromStr;
-use std::sync::{Arc, Mutex};
+use std::sync::{Arc, Mutex, mpsc};
```
Comment (benjaminbollen) [inline examples/simple_key_value_store.rs]: Can we now do without an Arc / Mutex anymore?

Comment (inetic) [inline examples/simple_key_value_store.rs]: I think we can.

File: examples/simple_key_value_store.rs [hunk 2]
```diff
@@ -139,294 +144,156 @@ struct CliArgs {
...
+}
 
-    fn handle_post_response(&mut self, _from_authority: Authority,
-        _from_address: types::SourceAddress, _response: ResponseError)
-        -> Vec<MethodCall> {
-        unimplemented!();
-    }
+impl Client {
+    fn new(_bootstrap_peers: Vec<Endpoint>) -> Result<Client, RoutingError> {
+        let (routing_sender, routing_receiver) = mpsc::channel::<Event>();
```
Comment (benjaminbollen) [inline examples/simple_key_value_store.rs]: just a style issue to consider: naming a channel WHAT_sender, and WHAT_receiver is more clear than WHO_sender/_receiver, as it feels to me less clear the direction of the channel

File: src/who_are_you.rs [hunk 3]
```diff
@@ -20,9 +20,9 @@ use public_id::PublicId;
 use types::Address;
```
Comment (dirvine) [inline src/who_are_you.rs]: We should rename the module / file here I think

Comment (benjaminbollen) [inline src/who_are_you.rs]: I'll rename it in #570

**Comment by benjaminbollen** [comment]
will already merge (in concert with @inetic s review of #570 - as both contain the others commits) 
the libsodium issue needs resolving independent from this PR

**Pull request merged**